src/main/java/org/openrewrite/maven/ResourceParser.java

```diff
@@ -90,8 +92,7 @@ public <S extends SourceFile> List<S> parseSourceFiles(
 
             long fileSize = attrs.size();
             if ((sizeThresholdMb > 0 && fileSize > sizeThresholdMb * 1024L * 1024L)) {
-                alreadyParsed.add(path);
-                logger.info("Skipping parsing " + path + " as its size + " + fileSize / (1024L * 1024L) +
+                logger.info("Parsing as Quark " + path + " as its size + " + fileSize / (1024L * 1024L) +
                         "Mb exceeds size threshold " + sizeThresholdMb + "Mb");
                 return false;
```

Comment:
You should parse it as a quark right here and add it to the `alreadyParsed` list or this message will be displayed n times, where "n" is the number of parsers that come before the quark parser.

Reply:
Alternatively, refactoring `ResourceParser` to not re-traverse the search directory for every single parser could also address this.